pkg/apis/sensor/v1alpha1/types.go

```diff
@@ -448,8 +448,9 @@ type URI struct {
 
 // ArtifactLocation describes the source location for an external artifact
 type ArtifactLocation struct {
-	S3     *S3Artifact `json:"s3,omitempty" protobuf:"bytes,1,opt,name=s3"`
-	Inline string      `json:"inline,omitempty" protobuf:"bytes,2,opt,name=inline"`
+	S3     *S3Artifact  `json:"s3,omitempty" protobuf:"bytes,1,opt,name=s3"`
+	Inline string       `json:"inline,omitempty" protobuf:"bytes,2,opt,name=inline"`
+	URL    *URLArtifact `json:"url,omitempty" protobuf:"bytes,4,opt,name=url"`
```

Comment:
The offset is intentionally set to 4. 3 will be used for file artifacts (https://github.com/argoproj/argo-events/pull/65).